corehq/apps/hqadmin/service_checks.py

```diff
@@ -0,0 +1,170 @@
+    except requests.ConnectionError:
+        return ServiceStatus(False, "Could not connect to touchforms")
+    else:
+        msg = "Touchforms returned a {} status code".format(res.status_code)
+        return ServiceStatus(res.ok, msg)
+
+
+@change_log_level('urllib3.connectionpool', logging.WARNING)
+def check_elasticsearch():
+    doc = {'_id': 'elasticsearch-service-check',
+           'date': datetime.datetime.now().isoformat()}
+    send_to_elasticsearch('groups', doc)
+    time.sleep(1)
+    hits = GroupES().remove_default_filters().doc_id(doc['_id']).run().hits
+    send_to_elasticsearch('groups', doc, delete=True)  # clean up
+    if doc in hits:
+        return ServiceStatus(True, "Successfully sent a doc to ES and read it back")
+    return ServiceStatus(False, "Something went wrong sending a doc to ES")
+
+
+def check_shared_dir():
+    return ServiceStatus(False, "Not implemented")
+
+
+def check_blobdb():
+    """Save something to the blobdb and try reading it back."""
+    db = get_blob_db()
+    contents = "It takes Pluto 248 Earth years to complete one orbit!"
+    info = db.put(StringIO(contents))
+    with db.get(info.identifier) as fh:
+        res = fh.read()
+    db.delete(info.identifier)
+    if res == contents:
+        return ServiceStatus(True, "Successfully saved a file to the blobdb")
+    return ServiceStatus(False, "Failed to save a file to the blobdb")
+
+
+def check_celery():
+    from celery import Celery
+    from django.conf import settings
+    celery = Celery()
+    celery.config_from_object(settings)
+    worker_responses = celery.control.ping(timeout=10)
+    if not worker_responses:
+        return ServiceStatus(False, 'No running Celery workers were found.')
+    else:
+        msg = 'Successfully pinged {} workers'.format(len(worker_responses))
+        return ServiceStatus(True, msg)
+
+
+def check_heartbeat():
+    celery_monitoring = getattr(settings, 'CELERY_FLOWER_URL', None)
+    if celery_monitoring:
+        cresource = Resource(celery_monitoring, timeout=3)
+        t = cresource.get("api/workers", params_dict={'status': True}).body_string()
+        all_workers = json.loads(t)
+        bad_workers = []
+        for hostname, status in all_workers.items():
+            if not status:
+                bad_workers.append('* {} celery worker down'.format(hostname))
+        if bad_workers:
+            return ServiceStatus(False, '\n'.join(bad_workers))
+
+    is_alive = heartbeat.is_alive()
+    return ServiceStatus(is_alive, "OK" if is_alive else "DOWN")
+
+
+def check_postgres():
+    a_user = User.objects.first()
+    if a_user is None:
+        return ServiceStatus(False, "No users found in postgres")
+    return ServiceStatus(True, "Successfully got a user from postgres")
+
+
+def check_couch():
+    """Confirm CouchDB is up and running, by hitting an arbitrary view."""
+    results = Application.view('app_manager/builds_by_date', limit=1).all()
```

Comment:
I think a lot of these checks would return success if they were run by the test runner. However, some of them have hard-coded details (like `app_manager/builds_by_date`) that could become outdated (for example, if this view were removed or renamed) and would return a false negative. What do you think of adding tests for these checks so we will know right away if some unrelated change breaks the check function? (can be done in a separate PR)